fintech-examples/fintech-ui/src/app/bank/redirect-page/redirect-page.component.ts

```diff
@@ -23,19 +25,22 @@ export class RedirectPageComponent implements OnInit {
   ngOnInit() {
     this.route.paramMap.subscribe(p => {
       this.redirectStruct = JSON.parse(p.get(HeaderConfig.HEADER_FIELD_LOCATION));
-      console.log('LOCATION IS ', this.redirectStruct.redirectUrl);
+      this.cardModal = {
+        title: 'Redirection',
+        description: 'Now we redirect you to your bank: ' + this.redirectStruct.bankName,
+        imageUrl: 'assets/icons/icons8-network 2.png',
+        confirmBtn: true,
+        cancelBtn: true
+      };
     });
   }
 
-  cancel(): void {
-    console.log('call from consent NOT ok for redirect ' + this.redirectStruct.redirectCode);
-    this.authService.fromConsentOk(Consent.NOT_OK, this.redirectStruct.redirectCode);
-  }
-
-  proceed(): void {
-    console.log('NOW GO TO:', decodeURIComponent(this.redirectStruct.redirectUrl));
-    // save user redirected state
-    this.storageService.isUserRedirected = true;
-    window.location.href = decodeURIComponent(this.redirectStruct.redirectUrl);
+  onSubmit(value: boolean) {
+    if (value) {
+      this.storageService.setUserRedirected(true);
+      window.location.href = decodeURIComponent(this.redirectStruct.redirectUrl);
```

Comment:
This wont work? It is important to send message with authService to FintechServer. Because in FintechServer redirect is already started and MUST be finished. With ok, or cancel. But if it is not called at all, garbage will remain.

Reply:
Further the authService.fromConsentOK does the resetting of the settings in the ui, which is important too.